src/python/turicreate/toolkits/sound_classifier/sound_classifier.py

```diff
@@ -105,14 +106,47 @@ def reset(self):
         # instance itself.
         pass
 
+class _NumPyDataIterator(_DataIterator):
+    def __init__(self, data, label=None, batch_size=1, shuffle=False):
+
+        # Always pass a tuple, so that the impl's built-in iterator returns a
+        # tuple.
+        self.data_slices = (data, label) if label is not None else (data,)
+        self.batch_size = batch_size
+        self.batch_idx = 0
+        self.num_batches = int(_ceil(len(data) / float(self.batch_size)))
+
+    def __iter__(self):
+        if self.batch_idx < self.num_batches:
+            data = self.data_slices[0][self.batch_size*self.batch_idx:self.batch_size*(self.batch_idx+1)]
+            label = None
+            if len(self.data_slices)>1:
+                label = self.data_slices[1][self.batch_size*self.batch_idx:self.batch_size*(self.batch_idx+1)]
+            self.batch_idx += 1
+            yield data, label if label is not None else data
+        else:
+            raise StopIteration
+
+    def reset(self):
+        # Each call to __iter__ returns a fresh iterator object that will do one
+        # pass through the data. The mxnet.io.NDArrayIter interface only needs
+        # this method because their __iter__ function returns the NDArrayIter
+        # instance itself.
+        self.batch_idx = 0
+
+
 def _create_data_iterator(data, label=None, batch_size=1, shuffle=False):
+
     if USE_TF:
-        return _TFDataIterator(data, label=label, batch_size=batch_size,
+        print("******** Using Numpy data iterator for TF")
```

Comment:
Might want to remove this.

Reply:
Yes, debugging print statements will be removed.

Reply:
In general I recommend committing all debugging code to a separate commit that you don't include in review branches. This keeps reviews a bit cleaner and unambiguous. There's some git tricks I can show you to make this easier....

Reply:
@nickjong yes please!